File: tree/108. 将有序数组转换为二叉搜索树.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
 var sortedArrayToBST = function(nums) {
    // 记录nums数组长度
    let n = nums.length;
    // 如果长度为零，返回null
    if(!n) return null;

    // 取向下取整的下标为中间坐标
    let mid = Math.floor(n / 2);
    // 创建节点，并赋值
    let root = new TreeNode(nums[mid]);
    // 如果mid为零,也就是数组长度为一，直接返回root
    if(mid ===0) return root;

    // 数组长度不为一，递归调用使用slice数组方法截取数组。
    root.left = sortedArrayToBST(nums.slice(0,mid))
    root.right = sortedArrayToBST(nums.slice(mid+1,n))
    // 返回root节点。
    return root;
};

File: Python/108. 将有序数组转换为二叉搜索树.py
# -*- codeing = utf-8 -*-
# @Time : 2023-8-18 21:58
# @Author : 刘永奇
# @File : 108. 将有序数组转换为二叉搜索树.py
# @Software : PyCharm
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        def makeTree(nums):
            if len(nums) == 0:
                return None
            mid = len(nums) // 2
            root = TreeNode(val = nums[mid])
            root.left = makeTree(nums[:mid])
            root.right = makeTree(nums[mid+1:])

            return root

        return makeTree(nums)
